src/Rector/Contrib/PHPUnit/SpecificMethod/AssertSameBoolNullToSpecificMethodRector.php

```diff
@@ -82,8 +80,19 @@ public function refactor(Node $methodCallNode): ?Node
 
     private function renameMethod(MethodCall $methodCallNode): void
     {
-        $newMethodName = $this->constValueToMethodNames[$this->costantName];
-        $methodCallNode->name = new Identifier($newMethodName);
+        /** @var Identifier $identifierNode */
+        $identifierNode = $methodCallNode->name;
+        $oldMethodName = $identifierNode->toString();
+
+        [$sameMethodName, $notSameMethodName] = $this->constValueToNewMethodNames[$this->constantName];
+
+        if ($oldMethodName === 'assertSame' && $sameMethodName) {
```

Comment:
@TomasVotruba If this PR gets accepted, I'll refactor `NodeChanger\MethodNameChanger` to support this `if/else` pattern that many `Rector`s have :sweat_smile: